index.js

```diff
@@ -123,6 +123,23 @@ const npmWalker = Class => class Walker extends Class {
       return super.onReaddir(entries)
     }
 
+    // when the cache has been seeded with the root manifest,
+    // we must respect that (it may differ from the filesystem)
+    const ig = path.resolve(this.path, 'package.json')
+
+    if (this.packageJsonCache.has(ig)) {
+      const pkg = this.packageJsonCache.get(ig)
+
+      // fall back to filesystem when seeded manifest is invalid
+      if (typeof pkg !== 'object' || Array.isArray(pkg)) {
```

Comment:
It feels weird to check for Array.isArray specifically here.  What if it's a Set, or a WeakMap, or `new SomeClass`?  (In fact, `null` will do the very wrong thing here, because null `typeof`'s to `object` and is not an array .)

I'd recommend either checking `Object.prototype.toString.call(pkg) === '[object Object]'`, or just treat any `typeof pkg === 'object' && pkg` as a valid value.  After all, we might want to represent package manifests as class instances someday.

Reply:
I think `Object.prototype.toString.call(pkg) !== '[object Object]'` is the ticket here, as it still allows custom class instances (with a proper `.toJSON()` method, of course) while _also_ guarding against the `null`/`undefined` I missed. Thanks!

Reply:
Turns out, if you pass `null` as the seeded value, it never reaches the conditional I added because it blows up the walker from `npm-bundled` first:
<img width="805" alt="image" src="https://user-images.githubusercontent.com/5605/73142656-186e6d80-4046-11ea-9533-b8df993b18e5.png">

`Object.prototype.toString.call(pkg)` is definitely the right call, regardless.